flow/core/kernel/scenario/base.py

```diff
@@ -0,0 +1,288 @@
+        with other sub-kernels.
+
+        Parameters
+        ----------
+        reset : bool
+            specifies whether the simulator was reset in the last simulation
+            step
+        """
+        raise NotImplementedError
+
+    def close(self):
+        """Close the scenario."""
+        raise NotImplementedError
+
+    ###########################################################################
+    #                        State acquisition methods                        #
+    ###########################################################################
+
+    def edge_length(self, edge_id):
+        """Return the length of a given edge/junction.
+
+        Return -1001 if edge not found.
+        """
+        raise NotImplementedError
+
+    def length(self):
+        """Return the total length of all junctions and edges."""
+        raise NotImplementedError
+
+    def speed_limit(self, edge_id):
+        """Return the speed limit of a given edge/junction.
+
+        Return -1001 if edge not found.
+        """
+        raise NotImplementedError
+
+    def max_speed(self):
+        """Return the maximum achievable speed on any edge in the network."""
+        raise NotImplementedError
+
+    def num_lanes(self, edge_id):
+        """Return the number of lanes of a given edge/junction.
+
+        Return -1001 if edge not found.
+        """
+        raise NotImplementedError
+
+    def get_edge_list(self):
+        """Return the names of all edges in the network."""
+        raise NotImplementedError
+
+    def get_junction_list(self):
+        """Return the names of all junctions in the network."""
+        raise NotImplementedError
+
+    def get_edge(self, x):  # TODO: maybe remove
+        """Compute an edge and relative position from an absolute position.
+
+        Parameters
+        ----------
+        x : float
+            absolute position in network
+
+        Returns
+        -------
+        edge position : tup
+            1st element: edge name (such as bottom, right, etc.)
+            2nd element: relative position on edge
+        """
+        raise NotImplementedError
+
+    def get_x(self, edge, position):  # TODO: maybe remove
+        """Return the absolute position on the track.
+
+        Parameters
+        ----------
+        edge : str
+            name of the edge
+        position : float
+            relative position on the edge
+
+        Returns
+        -------
+        absolute_position : float
+            position with respect to some global reference
+        """
+        raise NotImplementedError
+
+    def next_edge(self, edge, lane):
+        """Return the next edge/lane pair from the given edge/lane.
+
+        These edges may also be internal links (junctions). Returns an empty
+        list if there are no edge/lane pairs in front.
+        """
+        raise NotImplementedError
+
+    def prev_edge(self, edge, lane):
```

Comment:
Since we're redoing this right now, we may want to discuss what the appropriate thing to do is when two edges converge at a single edge

Reply:
sure! this is just architecturally what things will look like, but we can have this discussion when we start adding actual code